readthedocs/core/views/hooks.py

```diff
@@ -42,16 +42,26 @@ def _build_version(project, slug, already_built=()):
         # these will build at "latest", and thus won't be
         # active
         latest_version = project.versions.get(slug=LATEST)
-        trigger_build(project=project, version=latest_version, force=True)
-        log.info("(Version build) Building %s:%s",
-                 project.slug, latest_version.slug)
-        if project.versions.exclude(active=False).filter(slug=slug).exists():
-            # Handle the case where we want to build the custom branch too
-            slug_version = project.versions.get(slug=slug)
-            trigger_build(project=project, version=slug_version, force=True)
-            log.info("(Version build) Building %s:%s",
-                     project.slug, slug_version.slug)
-        return LATEST
+        if latest_version.active:
```

Comment:
The case where `latest` is disabled is missing.

Example:

```python
if latest.enabled:
  log.info('Building latest')  # note that it's important to log first then trigger
  trigger_build
else:
  log.info('Not building latest because disabled')
  return None
```

Reply:
> note that it's important to log first then trigger

I wanna to change that, but I wasn't sure if that was on purpose :thinking:

Reply:
The other logs do that too btw, so maybe we want to change those too

Reply:
Go ahead!

The idea of logging first is that if the following line fail for any reason, we miss the log and we don't know "where" it could have fail.